### PR TITLE
[receiver/purefa] implement fa_array_name to act as a pretty name as a label for metrics received. 

### PR DESCRIPTION
**Pull request by chrroberts-pure**
**Description:** 

As a user, I must be able correctly correlate metrics across arrays, to do so the metrics provided must have context as to which array they came from.

- It was decided from the Pure Storage Observability Practice that it will be best for the user to define it as a string to act as a pretty name as the label `fa_array_name`, or `host` if the array does not provide a `host` label. (e.g. if `host` is not provided as a label, the array is the host).

bugs fixed: the prometheus receiver that is used as a dependency does not support `.` in label names our implementation of adding labels - this may have effect in future planning for OTel Semantic conventions (e.g. host.name). This was causing the prom scraper to fail after metrics were collected in `receiver.go`

**Link to tracking Issue:** 
#23889
#21248
#22027

**Testing:** <Describe what testing was performed and which tests were added.>

- Tested with a live Pure Storage FlashArray, and verified labels were being added when exported via `logging`

- There is error validation on if fa_array_name == "", however there are some open test items in #23271 that are being resolved that this new pretty naming scheme should be included in.

**Documentation:** 
READMEs have been updated to be more verbose for how to scrape multiple arrays with multiple instances

### CODE REVIEW

File: receiver/purefareceiver/config.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ type Config struct {
 
 	// Env represents the respective environment value valid to scrape
 	Env string `mapstructure:"env"`
+
+	// Fa_array_name represents the display name that is appended to the received metrics, as the `host` label if not provided by OpenMetrics output, and to the `fa_array_name` label always.
+	Fa_array_name string `mapstructure:"fa_array_name"`
```
Comment (jpkrohling) [inline receiver/purefareceiver/config.go:45]: The var names should use camel case, and I think we can remove the FA prefix. Thus:

```suggestion
	ArrayName string `mapstructure:"fa_array_name"`
```

**Review by jpkrohling — COMMENTED** [review]
I'm OK in merging this, once the var name is changed to the same pattern as the others, but I would be way happier if this was accompanied by unit tests. This code has been a hotspot, it deserves its test.

**Comment by jpkrohling** [comment]
This PR is also missing a changelog entry: take a look at the files from .chloggen

File: receiver/purefareceiver/config.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ type Config struct {
 	Env string `mapstructure:"env"`
 
 	// Fa_array_name represents the display name that is appended to the received metrics, as the `host` label if not provided by OpenMetrics output, and to the `fa_array_name` label always.
-	Fa_array_name string `mapstructure:"fa_array_name"`
+	ArrayName string `mapstructure:"fa_array_name"`
```
Comment (jpkrohling) [inline receiver/purefareceiver/config.go:45]: You'll need to apply the rename elsewhere in this PR as well

Comment (chrroberts-pure) [reply receiver/purefareceiver/config.go:45]: This should be renamed where it needs to be. It was updated in config and receiver and tests

**Comment by chrroberts-pure** [comment]
@jpkrohling I have just rebased, and re-tested and I believe that this is ready to be merged. 

I also believe that the tests should be better, and I've added the new configurations in the test case, but I also believe it could be overhauled in a future release. I would like to go ahead and get this merged.

**Comment by chrroberts-pure** [comment]
I would also like to note that `fa_array_name` is an established label name across Observability platforms like Datadog

**Review by jpkrohling — APPROVED** [review]

**Comment by james-laing** [comment]
Tested with 0.83.0 and latest 0.88.0, working as expected.

**Pull request merged**